pkg/microservice/warpdrive/core/service/taskplugin/testing.go

```diff
@@ -124,6 +130,26 @@ func (p *TestPlugin) Run(ctx context.Context, pipelineTask *task.Task, pipelineC
 			return
 		}
 		p.kubeClient = kubeClient
+
+		clientset, err := kubeclient.GetClientset(pipelineTask.ConfigPayload.HubServerAddr, p.Task.ClusterID)
+		if err != nil {
+			msg := fmt.Sprintf("failed to get clientset: %s", err)
+			p.Log.Error(msg)
+			p.Task.TaskStatus = config.StatusFailed
+			p.Task.Error = msg
+			return
+		}
+		p.clientset = clientset
+
+		restConfig, err := kubeclient.GetRESTConfig(pipelineTask.ConfigPayload.HubServerAddr, p.Task.ClusterID)
+		if err != nil {
+			msg := fmt.Sprintf("failed to get clientset: %s", err)
+			p.Log.Error(msg)
+			p.Task.TaskStatus = config.StatusFailed
+			p.Task.Error = msg
+			return
+		}
+		p.restConfig = restConfig
```

Comment:
is it possible to extract the logic of init kubeclient/clientset/restcofig to a common function since it looks pretty much the same?

Reply:
It's actually better to use a public method. I'll change that.

Reply:
Maybe my last comment is  not clear, i mean the logic of initing  kubeclient&clientset&restcofig in `artifact_deploy.go` `testing.go` and `build.go` looks the same. Not the logic of initing kubeclient or clientset restcofig😁

Reply:
Yes, I abstracted this initialization into a method which is called by the three plugins.